test/performance/io/format_fasta_benchmark.cpp

```diff
@@ -0,0 +1,157 @@
+#include <cstring>
+
+#include <benchmark/benchmark.h>
+
+#include <sstream>
+
+#include <range/v3/algorithm/equal.hpp>
+#include <range/v3/view/transform.hpp>
+
+#include <seqan3/alphabet/quality/all.hpp>
+#include <seqan3/io/sequence_file/input_format_concept.hpp>
+#include <seqan3/io/sequence_file/output_format_concept.hpp>
+#include <seqan3/io/sequence_file/format_fasta.hpp>
+#include <seqan3/range/view/convert.hpp>
+
+using namespace seqan3;
+using namespace seqan3::literal;
+
+static void write(benchmark::State& state)
+{
+    std::stringstream ostream; //{"> seq\nACTAGACTAGCTACGATCAGCTACGATCAGCTACGA\n"};
+    sequence_file_format_fasta format;
+    sequence_file_output_options options;
+    std::string id{"seq"};
+    dna5_vector seq{"ACGTAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"_dna5};
+
+    for (auto _ : state)
+    {
+        format.write(ostream, options, seq, id, std::ignore);
+    }
+}
+
+BENCHMARK(write);
+
+#if __has_include(<seqan/seq_io.h>)
+
+    static void write2(benchmark::State& state)
+    {
+        seqan::CharString outStream;
+        seqan::CharString meta = "seq";
+        seqan::Dna5String seq = "ACGTAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+
+        for (auto _ : state)
+        {
+            seqan::writeRecord(outStream, meta, seq, seqan::Fasta());
+        }
+    }
+
+    BENCHMARK(write2);
+#endif
+
+static void read(benchmark::State& state)
+{
+    std::string dummy_file{};
+    for (size_t idx = 0; idx < 10000000; idx++)
+        dummy_file += ">seq\nACTAGACTAGCTACGATCAGCTACGATCAGCTACGA\n";
+
+    std::stringstream istream{dummy_file};
+    sequence_file_format_fasta format;
+    sequence_file_input_options<dna5, false> options;
+    std::string id;
+    dna5_vector seq;
+
+    for (auto _ : state)
+    {
+        format.read(istream, options, seq, id, std::ignore);
+    }
+}
+
+BENCHMARK(read);
+
+#if __has_include(<seqan/seq_io.h>)
+
+#include <fstream>
+
+std::fstream* createFastAFile(seqan::CharString &tempFilename)
+{
+    using namespace seqan;
+
+    tempFilename = SEQAN_TEMP_FILENAME();
+    char filenameBuffer[1000];
+    strncpy(filenameBuffer, toCString(tempFilename), 999);
+
+    std::fstream *file = new std::fstream(filenameBuffer, std::ios_base::in | std::ios_base::out | std::ios_base::binary | std::ios_base::trunc);
+    SEQAN_ASSERT(file->is_open());
+
+    std::string dummy_file{};
+    for (size_t idx = 0; idx < 10000000; idx++)
+        dummy_file += ">seq\nACTAGACTAGCTACGATCAGCTACGATCAGCTACGA\n";
+
+    char const * STR = dummy_file.c_str();
+
+    file->write(STR, strlen(STR));
+    file->seekg(0);
+    file->seekp(0);
+    return file;
+}
+
+    static void read2(benchmark::State& state)
+    {
+        seqan::CharString filename;
+        std::fstream *file = createFastAFile(filename);
+        seqan::SeqFileIn reader(*file);
+        seqan::CharString meta;
+        seqan::Dna5String seq;
+
+        for (auto _ : state)
+        {
+            seqan::readRecord(meta, seq, reader, seqan::Fasta());
```

Comment:
Same here. I am pretty sure that seqan2 readRecord also works with a string stream no?

Reply:
Keep The example with the file though and make one for seqan3 too